hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -874,12 +874,10 @@ private void positionThisBlockBuffer() {
       try {
         blockBuffer.skip(getCurCellSerializedSize());
       } catch (IllegalArgumentException e) {
-        LOG.error("Current pos = " + blockBuffer.position()
-            + "; currKeyLen = " + currKeyLen + "; currValLen = "
-            + currValueLen + "; block limit = " + blockBuffer.limit()
-            + "; currBlock currBlockOffset = " + this.curBlock.getOffset()
-            + "; path=" + reader.getPath());
-        throw e;
+        throw new IllegalArgumentException("Current pos = " + blockBuffer.position()
+            + "; currKeyLen = " + currKeyLen + "; currValLen = " + currValueLen + "; block limit = "
+            + blockBuffer.limit() + "; currBlock currBlockOffset = " + this.curBlock.getOffset()
+            + "; path=" + reader.getPath(), e);
```

Comment:
ok.

These patterns are written up? We should get them into our refguide under developer chapter.